core/src/apps/common/sdcard.py

```diff
@@ -120,9 +125,15 @@ async def ensure_sdcard(ensure_filesystem: bool = True) -> None:
 
             # Proceed to formatting. Failure is caught by the outside OSError handler
             with sdcard.filesystem(mounted=False):
-                fatfs.mkfs()
+                render_func = None
+                if not utils.DISABLE_ANIMATION:
+                    _start_progress()
+                    render_func = _render_progress
```

Comment:
I think for the sake of testing what is most similar to production it would make sense to set the callback even if it ultimately doesn't render anything. Then `DISABLE_ANIMATION` can be checked in `_start_progress` or perhaps in `RustProgress.report` if it doesn't break fixtures?

Reply:
https://github.com/trezor/trezor-firmware/pull/3700/commits/3b7460ec2e4ccb8b2b63d6f325fc058f06564f15